CONTRIBUTING.md

```diff
@@ -22,3 +36,19 @@ tox -e integration   # integration tests
 tox                  # runs 'lint' and 'unit' environments
 ```
 
+## Code overview
+
+TODO: expand on this section when spec doc is created
+
+## Intended use case
+
+TODO: expand on this section when spec doc is created
+
+## Roadmap
+
+TODO: expand on this section when spec doc is created
```

Comment:
Add the features we plan on adding that we don't already have, ie replication, sharding, backups, security :)